src/providers/Elsa.DistributedLocking.AzureBlob/AzureBlobLockProvider.cs

```diff
@@ -0,0 +1,199 @@
+using Elsa.DistributedLock;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.Azure.Storage.RetryPolicies;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Elsa.DistributedLocking.AzureBlob
+{
+    public class AzureBlobLockProvider : IDistributedLockProvider
+    {
+        private const string Prefix = "elsa";
+        private const string ContainerName = "elsa-lock-container";
+        private readonly AutoResetEvent _leaseSemaphore = new AutoResetEvent(true);
+        private static readonly object syncRoot = new Object();
+        private readonly List<LockedBlob> _lockedBlobs = new List<LockedBlob>();
+        private readonly string _connectionString;
+        private TimeSpan _leaseTime;
+        private readonly ILogger<AzureBlobLockProvider> _logger;
+        private CloudBlobContainer _cloudBlobContainer;
+        private readonly TimeSpan _renewInterval;
+        private Timer _renewTimer;
+
+        public AzureBlobLockProvider(string connectionString, TimeSpan leaseTime, TimeSpan renewInterval,
+                                        ILogger<AzureBlobLockProvider> logger)
+        {
+            _logger = logger;
+            _connectionString = connectionString;
+            if (leaseTime > TimeSpan.FromSeconds(60) || leaseTime < TimeSpan.FromSeconds(15))
+            {
+                _logger.LogInformation($"Leasetime must be between 15 Seconds and 60 seconds, Found {leaseTime.TotalSeconds} seconds. " +
```

Comment:
When creating log strings, you shouldn't use `$` for string interpolation. Instead, you should allow the logging system to perform the string interpolation itself. For example:

`_logger.LogInformation("Leasetime must be between 15 Seconds and 60 seconds, found {LeaseTimeTotalSeconds} seconds. Setting default value of 60 seconds", LeaseTime.TotalSeconds);`

See also: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-3.1#log-message-template

Please update the other logging statements as well. Thank you!

Reply:
Updated and checked-in.